### ISSUE TITLE
grouping could preserve original order

### ISSUE THREAD

**Issue by jangorecki** [h2oai/datatable#1070]
When doing
```
ans = x(select=mean(f.v1), groupby="id1")
```
we are getting aggregated results that are also sorted by `id1`. It is sometimes useful to retain original order of values from `id1` column, so aggregated results does not have to be sorted.

**Comment by st-pasha** [comment]
Yes, preserving the original order could be useful (this is the distinction of `by` and `keyby` in R). However, I believe this is not a very high priority right now, compared to other features we have on the plate.

**Comment by st-pasha** [comment]
A note to my future self (or anyone else who wants to tackle this issue) on how to implement this:
* Sort+groupby using the usual (non-order-preserving) method. This produces: a rowindex `ri1` (in which order the rows are to be taken), and the groupby `gby1` (where each group starts within the rowindex);
* Extract `ri1[gby1]` -- this will produce an array whose length is the number of groups;
* Sort this new array, again producing a rowindex describing the ordering of elements in the array. Call this rowindex `ri2`;
* Apply this new rowindex to the original rowindex, by group: 
  ```
  ri_res = sum(ri1[gby1[ri2[i]] : gby1[ri2[i]+1]] for i in len(ri2))
  ```
  The resulting groupby has to be computed by accumulating the sizes of groups in the new ordering.
